File: class 2/10845.py

class queue() :
    def __init__(self) :
        self.que= []
    
    def push(self, data) :
        self.que.append(data)

    def pop(self) :
        pop_object = None

        if self.isEmpty() :
            pop_object = -1     
        else :
            pop_object = self.que[0]
            self.que = self.que[1:]

        return pop_object

    def size(self) :
        return len(self.que)

    def isEmpty(self) :
        is_empty = False 
        if len(self.que) == 0 :
            is_empty = True
        
        return is_empty


    def empty(self) :
        is_empty = None
        if len(self.que) == 0 :
            is_empty = 1
        else :
            is_empty = 0
        
        return is_empty

    def front(self) :
        front_object = None

        if self.isEmpty() :
            front_object = -1
        else :
            front_object = self.que[0]
        
        return front_object

    def back(self) :
        back_object = None

        if self.isEmpty() :
            back_object = -1
        else :
            back_object = self.que[-1]
        
        return back_object
    
import sys    

num = int(sys.stdin.readline())
q = queue()

for i in range(num) :
    x = sys.stdin.readline().split()

    if x[0] == 'push' :
        q.push(int(x[1]))
    
    elif x[0] == "pop" :
        print(q.pop())

    elif x[0] == "size" :
        print(q.size())
    
    elif x[0] == "empty" :
        print(q.empty())
    
    elif x[0] == "front" :
        print(q.front())
    
    elif x[0] == "back" :
        print(q.back())
        
    

        

File: 백준 문제 풀이 모음/kotlin/class 2/10845.kt
import java.util.LinkedList
import java.util.Queue


val queue : Queue<Int> = LinkedList()

fun main () = with(System.`in`.bufferedReader()) {
    val n = readLine().toInt()
    val sb = StringBuilder()

    repeat(n) {
        val input = readLine().split(' ').toList()
        if (input.size == 2) {
            if (input[0].equals("push")) {
                queue.add(input[1].toInt())
            }
        } else {
            if (input[0].equals("pop")) {
                if (queue.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(queue.first())
                    queue.remove(queue.first())
                }
                sb.append('\n')
            } else if(input[0].equals("size")) {
                sb.append(queue.size)
                sb.append('\n')
            } else if(input[0].equals("empty")) {
                if (queue.isEmpty()) {
                    sb.append(1)
                } else {
                    sb.append(0)
                }
                sb.append('\n')
            } else if (input[0].equals("front")) {
                if (queue.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(queue.first())
                }
                sb.append('\n')
            } else if(input[0].equals("back")) {
                if (queue.isEmpty()) {
                    sb.append(-1)
                } else {
                    sb.append(queue.last())
                }
                sb.append('\n')
            }
        }
    }

    println(sb.toString())
}